src/main/scala/mesosphere/marathon/core/event/impl/callback/SubscribersKeeperActor.scala

```diff
@@ -38,38 +42,46 @@ class SubscribersKeeperActor(val store: EntityStore[EventSubscribers]) extends A
       subscription pipeTo sender()
 
     case GetSubscribers =>
-      val subscription = store.fetch(Subscribers).map(_.getOrElse(EventSubscribers()))(context.dispatcher)
-
+      val subscription = store.get().map(_.getOrElse(EventSubscribers()))(context.dispatcher)
       import context.dispatcher
       subscription pipeTo sender()
   }
 
   protected[this] def add(callbackUrl: String): Future[EventSubscribers] =
-    store.modify(Subscribers) { deserialize =>
-      val existingSubscribers = deserialize()
-      if (existingSubscribers.urls.contains(callbackUrl)) {
-        log.info("Existing callback {} resubscribed.", callbackUrl)
-        existingSubscribers
-      } else EventSubscribers(existingSubscribers.urls + callbackUrl)
-    }
+    lockManager.executeSequentially(LockName) {
+      async {
+        val subscribers = await(store.get()).getOrElse(EventSubscribers())
+        val updated = if (subscribers.urls.contains(callbackUrl)) {
+          log.info("Existing callback {} resubscribed.", callbackUrl)
+          subscribers
+        } else EventSubscribers(subscribers.urls + callbackUrl)
+
+        if (updated != subscribers) {
+          await(store.store(updated))
+        }
+        updated
+      }(ExecutionContext.global)
+    }(ExecutionContext.global) // blocks a thread, don't block the actor.
```

Comment:
just to be sure I understand:

CURRENT STATE: executeSequentially immediately blocks on obtaining the lock and only releases the lock once the `async {}` completes. executeSequentially may return before the lock is actually released.

FUTURE STATE: invoking executeSequentially will not block, and instead will queue an async operation that will be executed exclusive of all others sync'd to the same lock-name?